runtime/interpreter/decode.go

```diff
@@ -1755,6 +1762,77 @@ func (d *DecoderV4) decodeArrayElements(path []string) ([]Value, error) {
 	return values, nil
 }
 
+func (d *DecoderV4) decodeArrayValueStaticType(valuePath []string) (StaticType, error) {
+	const expectedLength = encodedArrayValueLength
```

Comment:
I was a bit confused by the function name and the variable declaration here, but I think I understand it now: This function decodes the start of an array value, in particular only the static type. Maybe I also misunderstood.

Maybe rename the function to make this more obvious (maybe `decodeArrayValueHead` or `decodeArrayValueStart`?), and add a comment describing the behaviour.

Reply:
yeah, this was a side effect of trying to re-use the same code. I'll see if I can split it into two to make it more intuitive, or else rename as you've suggestedd.

Reply:
Added in 9f54635